awscli/customizations/s3/filegenerator.py

```diff
@@ -157,7 +159,7 @@ def list_files(self, path, dir_op):
         if not self.should_ignore_file(path):
             if not dir_op:
                 size, last_update = get_file_stat(path)
-                yield path, size, last_update
+                yield path, size, last_update, None
```

Comment:
Yeah I was trying to touch the least amount of code paths as possible, but I agree it is getting too messy. I am going to just refactor the ``FileGenerator`` class to simplify some of the interfaces. Should not be hard.